undefined reference to `__cudaInitModule' when compiling run_hlo_module in debug mode

Build is configured with CUDA.

When compiling run_hlo_module as follows:
 
`
bazel build --config=dbg -j 24 --cxxopt="-D_GLIBCXX_USE_CXX11_ABI=0" //tensorflow/compiler/xla/tools:run_hlo_module
`

it results in the following error:

```
ERROR: /home/baarts/tensorflow/tensorflow/tensorflow/compiler/xla/tools/BUILD:96:1: Linking of rule '//tensorflow/compiler/xla/tools:replay_computation_gpu' failed (Exit 1)
bazel-out/k8-dbg/bin/external/nccl_archive/libdevice_a.a(sum_f64_reduce_scatter.cu.pic.o): In function `__nv_init_managed_rt_with_module(void**)':
/usr/local/cuda-10.2/bin/../targets/x86_64-linux/include/crt/host_runtime.h:264: undefined reference to `__cudaInitModule'
bazel-out/k8-dbg/bin/external/nccl_archive/libdevice_a.a(sum_f64_reduce.cu.pic.o): In function `__nv_init_managed_rt_with_module(void**)':
/usr/local/cuda-10.2/bin/../targets/x86_64-linux/include/crt/host_runtime.h:264: undefined reference to `__cudaInitModule'
bazel-out/k8-dbg/bin/external/nccl_archive/libdevice_a.a(sum_f64_all_reduce.cu.pic.o): In function `__nv_init_managed_rt_with_module(void**)':
/usr/local/cuda-10.2/bin/../targets/x86_64-linux/include/crt/host_runtime.h:264: undefined reference to `__cudaInitModule'
bazel-out/k8-dbg/bin/external/nccl_archive/libdevice_a.a(sum_f32_reduce_scatter.cu.pic.o): In function `__nv_init_managed_rt_with_module(void**)':
/usr/local/cuda-10.2/bin/../targets/x86_64-linux/include/crt/host_runtime.h:264: undefined reference to `__cudaInitModule'
bazel-out/k8-dbg/bin/external/nccl_archive/libdevice_a.a(sum_f32_reduce.cu.pic.o): In function `__nv_init_managed_rt_with_module(void**)':
/usr/local/cuda-10.2/bin/../targets/x86_64-linux/include/crt/host_runtime.h:264: undefined reference to `__cudaInitModule'
bazel-out/k8-dbg/bin/external/nccl_archive/libdevice_a.a(sum_f32_all_reduce.cu.pic.o):/usr/local/cuda-10.2/bin/../targets/x86_64-linux/include/crt/host_runtime.h:264: more undefined references to `__cudaInitModule' follow
collect2: error: ld returned 1 exit status
Target //tensorflow/compiler/xla/tools:replay_computation_gpu failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 65.233s, Critical Path: 55.19s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```

the rule that generates libdevice_a.a includes host_runtime.h which contains the following code:

```
static char __nv_init_managed_rt_with_module(void **handle)
{
  return __cudaInitModule(handle);
}
```

Because this is  a debug compilation, this function is not removed, and pulls in a reference to __cudaInitModule. This function is missing from tensorflow/stream_executor/cuda/cudart_stub.cc

The same error is there when compiling replay_computation_gpu and interactive_graphviz

--- a/tensorflow/stream_executor/cuda/cudart_stub.cc
+++ b/tensorflow/stream_executor/cuda/cudart_stub.cc
@@ -132,9 +132,9 @@ extern __host__ __device__ unsigned CUDARTAPI __cudaPushCallConfiguration(
 }
 
 extern char CUDARTAPI __cudaInitModule(void **fatCubinHandle) {
-  using FuncPtr = cudaError_t(CUDARTAPI *)(void **fatCubinHandle);
+  using FuncPtr = char(CUDARTAPI *)(void **fatCubinHandle);
   static auto func_ptr = LoadSymbol<FuncPtr>("__cudaInitModule");
-  if (!func_ptr) return GetSymbolNotFoundError();
+  if (!func_ptr) return 0;
   return func_ptr(fatCubinHandle);
 }
 

--- a/tensorflow/stream_executor/cuda/cudart_stub.cc
+++ b/tensorflow/stream_executor/cuda/cudart_stub.cc
@@ -131,6 +131,13 @@ extern __host__ __device__ unsigned CUDARTAPI __cudaPushCallConfiguration(
   return func_ptr(gridDim, blockDim, sharedMem, stream);
 }
 
+extern char CUDARTAPI __cudaInitModule(void **fatCubinHandle) {
+  using FuncPtr = cudaError_t(CUDARTAPI *)(void **fatCubinHandle);
+  static auto func_ptr = LoadSymbol<FuncPtr>("__cudaInitModule");
+  if (!func_ptr) return GetSymbolNotFoundError();
+  return func_ptr(fatCubinHandle);
+}
+
 #if CUDART_VERSION >= 10010
 extern void CUDARTAPI __cudaRegisterFatBinaryEnd(void **fatCubinHandle) {
   using FuncPtr = void(CUDARTAPI *)(void **fatCubinHandle);

